lib/rouster.rb

```diff
@@ -212,7 +212,15 @@ def initialize(opts = nil)
           @logger.debug(':instance specified, will connect to existing OpenStack instance')
           inst_details = self.ostack_describe_instance(@passthrough[:instance])
           raise ArgumentError.new(sprintf('No such instance found in OpenStack - %s', @passthrough[:instance])) if inst_details.nil?
-          @passthrough[:host] = inst_details.addresses["NextGen"][0]["addr"]
+          if defined?(inst_details.addresses["NextGen"][0]["addr"])
+            @passthrough[:host] = inst_details.addresses["NextGen"][0]["addr"]
+          else
+            inst_details.addresses.each_key do |address_key|
+              if defined?(inst_details.addresses[address_key][0]['addr'])
+                @passthrough[:host] = inst_details.addresses[address_key][0]['addr']
```

Comment:
The way we use the openstack api will just return a single key in 'addresses', since we bring up a single instance everytime. I have never seen a response that contains more than one addresses either. Will add a 'break' here just in case.